flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-elasticsearch/src/main/java/org/apache/flink/cdc/connectors/elasticsearch/serializer/ElasticsearchEventSerializer.java

```diff
@@ -156,6 +164,14 @@ private BulkOperationVariant createBulkOperationVariant(DataChangeEvent event)
         }
     }
 
+    public String tableSharding(TableId tableId, Map<String, Object> valueMap) {
+        if (shardingKey.containsKey(tableId)) {
```

Comment:
What about falling back to use the first partition column when partition columns was defined though `transform`